### PR TITLE
add mysql interpreter

### PR DESCRIPTION
**Pull request by 1teed**
Hi. I'm Hyun Roh

I added mysql interpreter.

Set interpreter
mysql_interpreter

How to use
mysql_interpreter_usage

### CODE REVIEW

**Comment by felixcheung** [comment]
Nice! Can this be generalized to all jdbc connections? It should work if the jar for the jdbc driver is there, right?

**Comment by swkimme** [comment]
Great job! 

BTW, why the test is failing on zeppelin-web? @Leemoonsoo @corneadoug Do you have some idea?

**Comment by Leemoonsoo** [comment]
Thanks for the contribution!
I think @felixcheung 's suggestion that generalize to jdbc really make sense.

File: mysql/src/main/java/com/nflabs/zeppelin/mysql/MysqlInterpreter.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    Connection conn;
+    Statement stmt;
+    ResultSet rs;
+
+    logger.info("Connect to " + url);
+
+    try {
+      // connect to mysql
+      Class.forName(driver);
+      conn = DriverManager.getConnection(url, user, password);
```
Comment (felixcheung) [inline mysql/src/main/java/com/nflabs/zeppelin/mysql/MysqlInterpreter.java]: should this be caching/pooling the connection?

Comment (felixcheung) [inline mysql/src/main/java/com/nflabs/zeppelin/mysql/MysqlInterpreter.java]: for example, override open/close()

**Comment by Leemoonsoo** [comment]
I have restarted test and strange build failure has gone.

**Comment by hyonaldo** [comment]
Such a nice thing!

**Comment by Leemoonsoo** [comment]
@1teed Zeppelin recently has moved to Apache and this repository will not accept update anymore. Could you recreate your pull-request to https://github.com/apache/incubator-zeppelin?

**Comment by bzz** [comment]
May be I'm missing something obvious here, but could you please also
explain in a new PR description what is the benefit of using separate
interpreter comparing to Spark SQL JDBC capabilities
<https://spark.apache.org/docs/1.3.0/sql-programming-guide.html#jdbc-to-other-databases>
?

On Sat, Mar 28, 2015 at 6:28 PM, Lee moon soo <notifications@github.com>
wrote:

> @1teed <https://github.com/1teed> Zeppelin recently has moved to Apache
> and this repository will not accept update anymore. Could you recreate your
> pull-request to https://github.com/apache/incubator-zeppelin?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NFLabs/zeppelin/pull/400#issuecomment-87197603>.
>



-- 
--
Kind regards,
Alexander.

**Comment by felixcheung** [comment]
@bzz It's a very good point, though to be fair, Spark 1.3.0 (first with DataFrame JDBC support) was just released a few days ago.
https://databricks.com/blog/2015/03/24/spark-sql-graduates-from-alpha-in-spark-1-3.html

**Comment by Leemoonsoo** [comment]
If i list out differences between mysql interpreter and JDBC DataFrame in spark,

* Mysql interpreter make connection from Zeppelin, while JDBC DataFrame make connection from all workers.
* JDBC DataFrame can manipulate query result on Spark, while Mysql interpreter is not.
* Let's say we visualize query result. With Mysql interpreter, Zeppelin just execute query and visualize the result. With JDBC DataFrame, create new JDBC DataFrame and register it as temptable and run SparkSQL over it.

I think Mysql interpreter make sense for people who don't need Spark.

**Comment by hyonaldo** [comment]
@bzz @felixcheung 
But for others who don't want to use Spark, Mysql interpreter will be a better way to use zeppelin for their own analysis system which has mysql only. (or Hive too)
Just because we can install and use Spark instead of Hive, it does not mean we don't need Hive interpreter at all.
I think zeppelin shouldn't force them to choose Spark only.

**Comment by riomario1** [comment]
When will this feature be available? After it is available, do I need to reinstall Zeppelin?

**Comment by Leemoonsoo** [comment]
We have moved to Apache. This pr is recreated https://github.com/apache/incubator-zeppelin/pull/60. Please check new PR.

To install, at least you'll need update binaries for interpreters(rebuild zeppelin) and update your configuration.

**Pull request closed without merging**